Sources/App/Commands/ReconcilePackageListCommand.swift

```diff
@@ -29,18 +29,20 @@ final class ReconcilePackageListCommand: Command
         
         // Delete any removed packages
         let packageDeletions = reconciler.packagesToDelete.map { url in
-          return Package.findByUrl(on: database, url: url).map { package -> Future<Void> in
+          return Package.findByUrl(on: database, url: url).flatMap { package -> Future<Void> in
```

Comment:
First of all, thank you so much for your help figuring this out @0xTim. You’re being incredibly generous with your time.

Now that you’ve pointed it out, I can see the problem. It’s the Future of a Future thing, again. 😬 But the fact that this fails in such a weird way, with the first DELETE statement getting as far as the database, but not actually deleting anything, and then everything else failing silently at both compile, and run time... it’s terrifying. 😱 I don't even know how I would have started to debug this myself.

I know this was my mistake and a fairly simple one at that, but this whole approach of throwing Futures into the void feels really fragile and opaque. Honestly, it scares me.

I think what I take from this is that using this kind of framework *every part of the system* needs to be tested. Nothing can be left to chance as the entire app is a typo away from silently breaking.

Reply:
Yeah it's a weird one and definitely a bit worrying that the compiler didn't pick it up, but the type inference and lack of using the futures masked it all. I must say this is the first time I've ever seen as issue like this be completely hidden